slack/plugin.py

```diff
@@ -306,3 +305,106 @@ def slack_conversations_history(
+    ) as client:
+        conversation_ids = _conversation_ids_by_name(client)
+        conversation_id = _find_conversation_by_name(conversation_ids, channel)
+        if conversation_id is None:
+            # `channel` does not match any name, so assume that it is an ID.
+            conversation_id = channel
+        params = {
+            "channel": conversation_id,
+            "ts": ts,
+            "limit": limit,
+        }
+        if cursor:
+            params["cursor"] = cursor
+        data = (
+            client.get(
+                "https://slack.com/api/conversations.replies",
+                params=params,
+            )
+            .raise_for_status()
+            .json()
+        )
+    if not data.get("ok", False):
+        if data.get("error") == "channel_not_found":
+            available_channels = f"{sorted(conversation_ids.keys())}"
+            # Avoid making the error message absurdly long.
+            max_length = 1024
+            if len(available_channels) > max_length:
+                truncated = "...(truncated)"
+                available_channels = (
+                    f"{available_channels[:max_length - len(truncated)]}{truncated}"
+                )
+            raise RuntimeError(
+                f"channel '{channel}' not found; "
+                f"available_channels: {available_channels}; "
+                "double-check that you have authorized the correct workspace"
+            )
+        raise RuntimeError(f"fetching replies: {data}")
+    messages = [
+        SlackMessage(
+            type=msg["type"],
+            user=msg.get("user"),
+            text=msg.get("text"),
+            ts=msg["ts"],
+        )
+        for msg in data.get("messages", [])
+    ]
+    next_cursor = data.get("response_metadata", {}).get("next_cursor", "")
+    return messages, next_cursor
+
+
+@purpose("Converts Slack user ID strings to display names")
+def slack_user_lookup(users: Sequence[str]) -> dict[str, SlackUser | None]:
+    """
+    Convert the sequence of slack user ID strings to a SlackUser object.
+
+    Args:
+      users: A sequence of strings corresponding to Slack user identifiers
+        like "U143B8CPZS5" (e.g., SlackMessage.user field).
+
+    Returns:
+      A mapping from the Slack user identifiers to the SlackUser object for
+        each input member, or None if not found.
+    """
+    with httpx.Client(
+        transport=AugmentedTransport(actions_v0.authenticated_request_slack)
+    ) as client:
+        # Get all users
+        all_users = _list_users(client)
+
+        user_set = set(users)
+        name_to_user: dict[str, SlackUser | None] = {}
+        # Initialize output to None for every user in the input.
+        for input_user in user_set:
+            name_to_user[input_user] = None
+
+        for slack_user in all_users:
+            if slack_user.id in user_set:
+                name_to_user[slack_user.display_name] = slack_user
```

Comment:
```suggestion
                name_to_user[slack_user.id] = slack_user
```
Should this not be the `id`, because `display_name` is a different thing?